harness/attributes/docker-base.yml

```
@@ -63,13 +63,39 @@ attributes:
       resources:
         memory: "1024Mi"
     redis:
-      image: redis:5-alpine
+      image: redis:6-alpine
+      options:
+        # Handle many smaller keys
+        activedefrag: 'yes'
+        # 1Gi - 1024*1024*1024 bytes
+        maxmemory: '1073741824'
+        # Evict any least recently used key even if they don't have a TTL
+        maxmemory-policy: allkeys-lru
+        # Save snapshots every X changes have happened within Y seconds (these are the defaults in redis.conf)
+        save:
+          - 3600 1
+          - 300 100
+          - 60 10000
```

```suggestion
        save: "3600 1 300 100 60 10000"
```
Redis 7 doesn't like this options style

---

original was:
```
--save 3600 1 --save 300 100 -- save 60 10000
```

implementation made:
```
'--save 3600 1' '--save 300 100' '--save 60 10000'
```

suggestion made:
```
'--save 3600 1 300 100 60 10000'
```

expected:
```
--save 3600 1 --save 300 100 -- save 60 10000
```

which is:
```
'--save' '3600' '1' '--save' '300' '100' '--save' '60' '10000'
```

---

though one question is, does redis think the command is ok even when no real option arguments but in the option?

---

I made a PR that I think does the right quoting for < 7 https://github.com/inviqa/harness-docker/pull/85

---

"these are the defaults in redis.conf"

so I guess we could avoid supplying them by default rather than provide 2 different config styles toggled by version

---

can document the two save styles as comments I guess

A bit annoying tho the complexity of save warrants it's own tests